app/src/main/java/com/sanjeevyadavit/magecart/data/remote/dto/product/ProductDto.kt

```diff
@@ -36,4 +38,18 @@ data class ProductData(
     val updatedAt: String,
     @SerializedName("visibility")
     val visibility: Int
-)
\ No newline at end of file
+)
+
+fun ProductDto.toProduct(): Product {
+    val thumbnailUrl = customAttributes.find {
+        it.attributeCode == Constants.THUMBNAIL_SK
+    }?.value as String
```

Comment:
if find will return null, then it would crash. you should do null safe typecasting something like this
```
val thumbnailUrl = customAttributes.find {
        it.attributeCode == Constants.THUMBNAIL_SK
    }?.value as? String
```
or 
```
val thumbnailUrl = customAttributes.find {
        it.attributeCode == Constants.THUMBNAIL_SK
    }?.value.orEmpty()
```